public/app/plugins/panel/canvas/editor/element/utils.ts

```diff
@@ -8,6 +8,11 @@ import { HttpRequestMethod } from '../../panelcfg.gen';
 import { APIEditorConfig } from './APIEditor';
 
 export const callApi = (api: APIEditorConfig, isTest = false) => {
+  // If API endpoint origin matches Grafana origin, don't call it.
+  if (requestMatchesGrafanaOrigin(api.endpoint)) {
+    appEvents.emit(AppEvents.alertError, ['Cannot call API at Grafana origin.']);
+    return;
+  }
   if (api && api.endpoint) {
```

Comment:
Should this check be inside the check verifying the api object + endpoint property exist?

Reply:
Yeah, good call. Undefined will fail.